engine/lib/phx/src/input/devices/gamepad/state.rs

```diff
@@ -0,0 +1,254 @@
+                        // let button_settings = gamepad_settings.get_button_axis_settings(button);
+
+                        // // Only send events that pass the user-defined change threshold
+                        // if let Some(filtered_value) = button_settings.filter(raw_value, old_value) {
+                        //     events.send(
+                        //         GamepadButtonChangedEvent::new(
+                        //             gamepad_id,
+                        //             button_type,
+                        //             filtered_value,
+                        //         )
+                        //         .into(),
+                        //     );
+                        // }
+                    }
+                }
+                EventType::AxisChanged(gilrs_axis, raw_value, _) => {
+                    if let Some(axis) = convert_axis(gilrs_axis) {
+                        if let Some(state) = self.device_state.get_mut(&gamepad_id) {
+                            state.axis_state.update(axis as _, raw_value);
+
+                            if state.control_state.update() {
+                                res = Some(gamepad_id);
+                            }
+                        }
+
+                        // TODO: threshold check?
+                        // let axis = GamepadAxis::new(gamepad_id, axis_type);
+                        // let old_value = gamepad_axis.get(axis);
+                        // let axis_settings = gamepad_settings.get_axis_settings(axis);
+
+                        // // Only send events that pass the user-defined change threshold
+                        // if let Some(filtered_value) = axis_settings.filter(raw_value, old_value) {
+                        //     events.send(
+                        //         GamepadAxisChangedEvent::new(gamepad_id, axis_type, filtered_value)
+                        //             .into(),
+                        //     );
+                        // }
+                    }
+                }
+                _ => (),
+            };
+        }
+
+        self.gilrs.inc();
+
+        res
+    }
+}
+
+#[luajit_ffi_gen::luajit_ffi]
+impl GamepadState {
+    pub fn gamepads_count(&self) -> usize {
+        self.device_state.len()
+    }
+
+    pub fn gamepad_id(&self, index: usize) -> Option<GamepadId> {
+        let device_ids: Vec<_> = self.device_state.keys().collect();
+
+        device_ids.get(index).map(|id| **id)
+    }
+
+    pub fn gamepad_name(&self, gamepad_id: GamepadId) -> Option<String> {
+        self.device_state
+            .get(&gamepad_id)
+            .map(|state| state.name.clone())
+    }
+
+    pub fn value(&self, axis: GamepadAxis) -> f32 {
+        self.device_state
+            .iter()
+            .find_map(|(_, state)| {
+                state
+                    .control_state
+                    .is_connected()
+                    .then(|| state.axis_state.value(axis as _)) // TODO: button value
```

Comment:
Do we need to update this?

Reply:
Not needed. Removed. Any issues with gamepad can be fixed later during proper testing.